src/emeus-constraint.c

```diff
@@ -612,3 +631,35 @@ emeus_constraint_is_attached (EmeusConstraint *constraint)
 
   return constraint->solver != NULL;
 }
+
+void
+emeus_constraint_set_active (EmeusConstraint *constraint,
+                             gboolean         active)
+{
+  g_return_if_fail (EMEUS_IS_CONSTRAINT (constraint));
+
+  active = !!active;
```

Comment:
Yes; `gboolean` is really a typedef to `int`, which means that — unlike C99's `_Bool` — the compiler will not coerce the allowed values to 0 or 1.

In general, the `val = !!val` self assignment is required when storing a `gboolean` argument into a bit field — like `guint is_val : 1` — as values bigger than the storage size have undefined behaviour, but it's good code hygiene to do it anyway.